diff.py

```diff
@@ -1786,12 +1786,24 @@ def _normalize_adrp_differences(self, mnemonic: str, row: str) -> str:
 
 class AsmProcessorI686(AsmProcessor):
     def process_reloc(self, row: str, prev: str) -> Tuple[str, Optional[str]]:
+        if "WRTSEG" in row: # ignore WRTSEG (watcom)
+            return prev, None
         repl = row.split()[-1]
         mnemonic, args = prev.split(maxsplit=1)
-
-        addr_imm = re.search(r"(?<!\$)0x[0-9a-f]+", args)
+        # Example 0x0,0x8(%edi)
+        # Example 0x0,%edi
+        # Example *0x0(,%edx,4)
+        # Match 0x0 part to replace
```

Comment:
Do you get 0x0 even if the reloc has an addend, e.g., when compiling
```
int x[2];
int f() { return x[1]; }
int* g() { return &x[1]; }
```
?

Reply:
Okay, I've refactored the reloc handling again to handle those kinds of cases, any other cases you can think of that I might need to handle?